spinta/manifests/open_api/helpers.py

```diff
@@ -2,99 +2,353 @@
 
 import json
 import re
+from collections.abc import Generator
 from pathlib import Path
-from typing import Generator
 
 from spinta.core.ufuncs import Expr
-from spinta.utils.naming import to_dataset_name, to_code_name
+from spinta.utils.naming import Deduplicator, to_code_name, to_dataset_name, to_model_name, to_property_name
 
-
-SUPPORTED_PARAMETER_LOCATIONS = {'query', 'header', 'path'}
-DEFAULT_DATASET_NAME = 'default'
+SUPPORTED_PARAMETER_LOCATIONS = {"query", "header", "path"}
+DEFAULT_DATASET_NAME = "default"
+SCHEMA_REF_KEY = "$ref"
+DEFAULT_PROPERTY_DATATYPE = "string"
 
 
 def replace_url_parameters(endpoint: str) -> str:
     """Replaces parameters in given endpoint to their codenames.
 
     e.g. /api/cities/{cityId}/ -> /api/cities/{city_id}
     """
-    return re.sub(r'{([^{}]+)}', lambda match: f'{{{to_code_name(match.group(1))}}}', endpoint)
+    return re.sub(r"{([^{}]+)}", lambda match: f"{{{to_code_name(match.group(1))}}}", endpoint)
 
 
 def read_file_data_and_transform_to_json(path: Path) -> dict:
     with Path(path).open() as file:
         return json.load(file)
 
 
+def resolve_refs(schema: dict, root: dict, seen: set[str] = None) -> dict:
```

Comment:
It shouldn't replace refs with their counterpart schemas. As we talked, this might create duplicate models where they aren't needed. Maybe check how it's done in the XSD schema parsing. It's a bit complex, but it solves all the issues without creating duplicates. 

I'd suggest to move `ref` handling to a different task (create a new ticket) and a different PR.

Reply:
I would suggest not to handle ref here at all, but do it properly from the beginning